hail/python/hail/experimental/ldscsim.py

```diff
@@ -82,43 +85,51 @@ def simulate_phenotypes(mt, genotype, h2, pi=None, rg=None, annot=None, popstrat
     popstrat: :class:`.Expression`, optional
         Column field to use as our aggregated covariates for adding population
         stratification.
-    
+    exact_h2: :obj:`bool`, optional
+        Whether to exactly simulate ratio of variance of genetic component of 
+        phenotype to variance of phenotype to be h2. If `False`, ratio will be
+        h2 in expectation. Observed h2 in the simulation will be close to 
+        expected h2 for large-scale simulations.
+
     Returns
     -------
     :class:`.MatrixTable`
         :class:`.MatrixTable` with simulated betas and phenotypes, simulated according
         to specified model.
     """
-    h2 = h2.tolist() if type(h2) is np.ndarray else ([h2] if type(h2) is not list else h2) 
+    h2 = h2.tolist() if type(h2) is np.ndarray else ([h2] if type(h2) is not list else h2)
     pi = pi.tolist() if type(pi) is np.ndarray else pi
-    tid = ''.join(random.choices(string.ascii_uppercase + string.ascii_lowercase, k=5)) # "temporary id" -- random string to identify temporary intermediate fields generated by this method
+    # tid = "temporary id" -- random string to identify temporary intermediate fields generated by this method
+    tid = _get_tid()
```

Comment:
you can use `hl.utils.java.Env.get_uid()`, which we do internally in the library.

Reply:
oh cool, I should have asked if you already had something like this